Data/ByteString/Lazy.hs

```diff
@@ -953,6 +954,19 @@ findIndex k cs0 = findIndex' 0 cs0
             Just i  -> Just (n + fromIntegral i)
 {-# INLINE findIndex #-}
 
+-- | The 'findIndexEnd' function takes a predicate and a 'ByteString' and
+-- returns the index of the last element in the ByteString
+-- satisfying the predicate.
+findIndexEnd :: (Word8 -> Bool) -> ByteString -> Maybe Int64
+findIndexEnd k = findIndexEnd' 0
+  where
+    findIndexEnd' _ Empty = Nothing
+    findIndexEnd' n (Chunk c cs) =
+      let !n' = n + S.length c
+          !i  = fmap (fromIntegral . (n +)) $ S.findIndexEnd k c
+      in findIndexEnd' n' cs `mplus` i
+{-# INLINE findIndexEnd #-}
```

Comment:
I'm not sure how efficient this is, but it's very similar to the existing `elemIndexEnd`, so at least it shouldn't be worse.

https://github.com/haskell/bytestring/blob/95fe6bdf13c9cc86c1c880164f7844d61d989574/Data/ByteString/Lazy.hs#L909-L925